4/index.php

```diff
@@ -1,136 +1,68 @@
 <?php
+declare(strict_types=1);
 
-namespace docker {
+if (basename($_SERVER['REQUEST_URI']) === 'adminer.css' && is_readable('adminer.css')) {
+    header('Content-Type: text/css');
+    readfile(__DIR__ . '/adminer.css');
+    exit;
+}
 
-    class ConfigProvider
+function adminer_object(): Adminer
+{
+    return new class extends Adminer
     {
-        /** @return array<int, ?string> */
-        public static function getLoginConfig(): array
+        public function name(): ?string
         {
-            return [
-                'db' => self::getEnv('ADMINER_DB'),
-                'driver' => self::getEnv('ADMINER_DRIVER'),
-                'password' => self::getEnv('ADMINER_PASSWORD'),
-                'server' => self::getEnv('ADMINER_SERVER'),
-                'username' => self::getEnv('ADMINER_USERNAME'),
-                'name' => self::getEnv('ADMINER_NAME'),
-            ];
+            return $this->getEnv('ADMINER_NAME') ?? parent::name();
         }
 
-        /**
-         * For Adminer::name()
-         */
-        public static function getAdminerName(): ?string
+        public function loginForm(): void
         {
-            return self::getEnv('ADMINER_NAME');
+            parent::loginForm();
+
+            if ($this->getEnv('ADMINER_AUTOLOGIN')) {
+                echo script('
+                    document.addEventListener(\'DOMContentLoaded\', function () {
+                        document.forms[0].submit()
+                    })
+                ');
+            }
         }
 
-        public static function getLoginConfigSingle(string $key): ?string
+        public function loginFormField($name, $heading, $value): string
         {
-            return self::getLoginConfig()[$key] ?? null;
-        }
+            $envValue = $this->getLoginConfigValue($name);
+
+            if ($envValue !== null) {
+                $value = sprintf(
+                    '<input name="auth[%s]" type="%s" value="%s">',
```

Comment:
The original implementation didn't leak password to the end user (AFAIK), that was intended feature.